### PR TITLE
solved lesson8

### PR DESCRIPTION
**Pull request by kinuesantos**

### CODE REVIEW

File: test/lesson8.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const findMax= require('../solutions/lesson8.js');
+const test = (obj,result)=>{
+  if(findMax(obj)==result){
+    console.log(`correct, the result is ${findMax(obj)}`);
+  }else{
+    console.log(`wrong the result is ${result}`);
+  }
+}
+const a = {val:64,next:{val:75,next:{val:90,next:null}}};
```
Comment (allopez7) [inline test/lesson8.js]: Add a test case object with all negative values.

File: solutions/lesson8.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const findMax=(obj,max=0)=>{
+  if(obj.next==null){
```
Comment (saumya665) [inline solutions/lesson8.js]: Also check for circular linked lists.

File: solutions/lesson8.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const findMax=(obj,max=0)=>{
```
Comment (saumya665) [inline solutions/lesson8.js]: create a new parameter for the head. head = linkedL

File: solutions/lesson8.js (hunk 2 above)
Comment (saumya665) [inline solutions/lesson8.js]: also check if obj.next==head

File: solutions/lesson8.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  if(obj.next==null){
+    if(obj.val>max){
+        max=obj.val;
+    }
+  return max;
+  }
+  if(obj.val>max){
```
Comment (saumya665) [inline solutions/lesson8.js]: repeating code?

File: solutions/lesson8.js (hunk 3 above)
Comment (kinuesantos) [reply solutions/lesson8.js]: done

File: test/lesson8.js (hunk 1 above)
Comment (kinuesantos) [reply test/lesson8.js]: all done, thankss! :)

File: solutions/lesson8.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const findMax=(obj,head=obj,max=obj.val)=>{
+  if(obj.val>max){
+    max = obj.val;
+  }
+if(obj.next==null || obj.next==head){
```
Comment (songz) [inline solutions/lesson8.js]: null evaluates to false. so you can simply do `if (!obj.next || ...`

File: test/lesson8.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+}
+const a = {val:-64,next:{val:-75,next:{val:-90}}};
+const b = {val:52,next:{val:66}};
+
+const c = {val:32}
+d={val:33}
+e={val:34}
+c['next']=d;
+d['next']=e;
+e['next']=c
```
Comment (seemcat) [inline test/lesson8.js]: you forgot a ; 

pleaseeeee get yo syntax & formatting right
to reformat code:
**gg** to get to the top of the page
**G** to reformat the whole page

File: solutions/lesson8.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const findMax=(obj,head=obj,max=obj.val)=>{
+  if(obj.val>max){
+    max = obj.val;
+  }
```
Comment (seemcat) [inline solutions/lesson8.js]: if (obj.val > max) max = obj.val;

File: solutions/lesson8.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+if(!obj.next || obj.next==head){
+  return max;
+}
```
Comment (seemcat) [inline solutions/lesson8.js]: if (!obj.next || obj.next == head) return max;

**Review by seemcat — CHANGES_REQUESTED** [review]

**Review by seemcat — APPROVED** [review]
